### PR TITLE
Add tests for Canvas2D imageSmoothingEnabled

### PR DESCRIPTION
**Pull request by Elchi3**

### CODE REVIEW

**Comment by tobie** [comment]
Reviewers for this pull request are: @w3c/2dcontext-reviewers.

**Comment by Elchi3** [comment]
@plehegar I addressed the two comments.

**Comment by Ms2ger** [comment]
@sideshowbarker, can you try to get this reviewed and landed?

**Comment by sideshowbarker** [comment]
@junov any chance you got time to review this?

**Comment by sideshowbarker** [comment]
This branch has a merge conflict that I don’t yet understand how to resolve:
```
$ git pull https://github.com/Elchi3/web-platform-tests.git imageSmoothingEnabled
From https://github.com/Elchi3/web-platform-tests
 * branch            imageSmoothingEnabled -> FETCH_HEAD
CONFLICT (file/directory): There is a directory with name tools in ebc53e7d4cc4695ca6483eeca4abf74d2a773b6b. Adding tools as tools~HEAD
Auto-merging quirks-mode/percentage-height-calculation.html
Automatic merge failed; fix conflicts and then commit the result.
Error redoing merge bd10f223822a24de20e12fce1b3cde3d4998176e
```

I guess that has something to do with the changes we made to `tools` in the time since the branch for this PR was originally created. But I don’t know how to resolve it.

File: 2dcontext/the-canvas-state/2d.state.saverestore.imageSmoothingEnabled.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  imgctx.fillStyle = "#F00";
+  imgctx.fillRect(0, 0, 2, 2);
+  imgctx.fillStyle = "#0F0";
+  imgctx.fillRect(0, 0, 1, 1);
+  return image;
+}
+
+test(function() {
+  var ctx = document.createElement('canvas').getContext('2d');
+  var old = ctx.imageSmoothingEnabled;
```
Comment (junov) [inline 2dcontext/the-canvas-state/2d.state.saverestore.imageSmoothingEnabled.html]: Instead havin 'old', the test should assert that imageSmoothingEnabled is initially true;

Comment (sideshowbarker) [inline 2dcontext/the-canvas-state/2d.state.saverestore.imageSmoothingEnabled.html]: > Instead having 'old', the test should assert that imageSmoothingEnabled is initially true;

@Elchi3 can you make that change?

**Comment by sideshowbarker** [comment]
@Elchi3 To help resolve the merge conflict without needing to create a new PR, I created the https://github.com/w3c/web-platform-tests/tree/Elchi3/imageSmoothingEnabled branch off of current master here with your changes. So if you want to, you could just pull that branch and then force-push it to the `imageSmoothingEnabled` branch in your fork. So that will update this PR and allow us to merge it to master here (after the review change from @junov gets made).

**Comment by Elchi3** [comment]
Updated.

**Comment by sideshowbarker** [comment]
@Elchi3 Thanks yeah I see no more merge conflicts and I see you made @junov’s review change.

@junov, if you can give a LGTM for this change overall in its current state, I’ll go ahead and merge it.

**Comment by junov** [comment]
lgtm

**Comment by sideshowbarker** [comment]
@junov @Elchi3 thanks!

**Pull request merged**